packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.spec.ts

```diff
@@ -0,0 +1,385 @@
+						.join("")
+				);
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(htmlInputElement.value).toBe(expected);
+		});
+	});
+
+	describe("on textarea", () => {
+		@Component({
+			selector: "test-component",
+			template: "<textarea [starkTransformInput]='starkTransformInputValue'></textarea>"
+		})
+		class TestComponent {
+			public starkTransformInputValue: StarkInputTransformationType = () => {
+				/*noop*/
+			};
+		}
+
+		let fixture: ComponentFixture<TestComponent>;
+		let component: TestComponent;
+		let htmlInputElement: HTMLTextAreaElement;
+
+		beforeEach(() => {
+			TestBed.configureTestingModule({
+				declarations: [StarkTransformInputDirective, TestComponent]
+			});
+
+			fixture = TestBed.createComponent(TestComponent);
+			component = fixture.componentInstance;
+			htmlInputElement = fixture.nativeElement.querySelector("textarea");
+			htmlInputElement.value = "";
+
+			// trigger initial data binding
+			fixture.detectChanges();
+
+			expect(htmlInputElement.value).toBe("", "field 'value' should start as empty string ");
+		});
+
+		it("should set value to uppercase", () => {
+			const input: string = "upper";
+			const expected: string = "UPPER";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "uppercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(htmlInputElement.value).toBe(expected);
+		});
+
+		it("should set value to lowercase", () => {
+			const input: string = "LOWER";
+			const expected: string = "lower";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = "lowercase";
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(htmlInputElement.value).toBe(expected);
+		});
+
+		it("should replace dirty word", () => {
+			const input: string = "fudge you!";
+			const expected: string = "***** you!";
+
+			// Set directive to correct implementation
+			component.starkTransformInputValue = (v: string) =>
+				v.replace("fudge", (match: string) =>
+					match
+						.split("")
+						.map(() => "*")
+						.join("")
+				);
+			fixture.detectChanges();
+
+			mockInputEvent(input, htmlInputElement);
+			fixture.detectChanges();
+
+			expect(htmlInputElement.value).toBe(expected);
+		});
+	});
+
+	describe("invalid input", () => {
+		@Component({
+			selector: "test-component",
+			template: "<input starkTransformInput='InvalidInput'/>"
+		})
+		class TestComponent {}
+
+		beforeEach(() => {
+			TestBed.configureTestingModule({
+				declarations: [StarkTransformInputDirective, TestComponent]
+			});
+		});
+
+		it("should throw an error", () => {
+			const fixture: ComponentFixture<TestComponent> = TestBed.createComponent(TestComponent);
+			// trigger initial data binding
+			expect(() => fixture.detectChanges()).toThrowError();
```

Comment:
Can you add a regex to the `toThrowError` to check the message of the error? It should not be too specific, something generic that matches the error so we know it is indeed the error we manually throw